onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -89,6 +91,46 @@ bool operator!=(const ONNX_NAMESPACE::TensorShapeProto_Dimension& l,
   return !(l == r);
 }
 
+#ifdef _WIN32
+namespace checker {
+std::wstring resolve_external_data_location(
+    const std::wstring& base_dir,
+    const std::wstring& location,
+    const std::string& tensor_name) {
+  auto file_path = std::filesystem::path(location);
+  if (file_path.is_absolute()) {
+    fail_check(
+        "Location of external TensorProto ( tensor name: ",
+        tensor_name,
+        ") should be a relative path, but it is an absolute path: ",
+        wstring_to_utf8str(location));
+  }
+  auto relative_path = file_path.lexically_normal().make_preferred().wstring();
+  // Check that normalized relative path contains ".." on Windows.
+  if (relative_path.find(L"..", 0) != std::string::npos) {
+    fail_check(
+        "Data of TensorProto ( tensor name: ",
+        tensor_name,
+        ") should be file inside the ",
+        wstring_to_utf8str(base_dir),
+        ", but the '",
+        wstring_to_utf8str(location),
+        "' points outside the directory");
+  }
+  std::wstring data_path = onnxruntime::ConcatPathComponent(base_dir, relative_path);
+  struct _stat64 buff;
+  if (data_path.empty() || (data_path[0] != '#' && _wstat64(data_path.c_str(), &buff) != 0)) {
```

Comment:
You may consider using std::filesystem::exists() instead, so that we would not mix the uses of old C-style IO functions and the newer STL ones.

Reply:
there are pipeline that are still use pre c++17. will we upgrade all CIs before next release this function does not have to be written in such a way that std::filesystem used depending on build flags:
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1311817&view=logs&j=b3525746-07e2-593d-e19f-f545f35bd083&t=0a2808e4-f410-59d3-9ece-0d76a5245e20